05_stringer/luliangLi/main_test.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestStringify(t *testing.T) {
+	testCases := map[string]struct {
+		input    IPAddr
+		expected string
+	}{
+		"localhost": {input: IPAddr{127, 0, 0, 1}, expected: "127.0.0.1"},
+		"google":    {input: IPAddr{8, 8, 8, 8}, expected: "8.8.8.8"},
+		"blank":     {input: IPAddr{}, expected: "0.0.0.0"},
+		"quater":    {input: IPAddr{127}, expected: "127.0.0.0"},
+		"half":      {input: IPAddr{127, 1}, expected: "127.1.0.0"},
+		"3quaters":  {input: IPAddr{127, 1, 1}, expected: "127.1.1.0"},
+	}
+	for name, test := range testCases {
+		input, expected := test.input, test.expected
+		t.Run(name, func(t *testing.T) {
+			actual := input.String()
+			if expected != actual {
+				t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
+			}
+		})
+	}
+}
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("127.0.0.1\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
```

Comment:
Identically error message to @n0npax's [`t.Errorf`](https://github.com/anz-bank/go-course/pull/215/files#diff-333f80df5d7b211e6addaa175bd579d7R21)